meteor/sofie-server/data-access/repositories/test/mongo-part-repository.spec.ts

```diff
@@ -0,0 +1,174 @@
+			const part: Part = createPart({ segmentId: segmentId })
+			await testDatabase.populateDatabaseWithParts([part])
+			const db: Db = testDatabase.getDatabase()
+
+			when(mongoConverter.convertParts(anything())).thenReturn([part])
+			const testee: PartRepository = await createTestee(db, {
+				mongoConverter: mongoConverter,
+			})
+
+			await testee.deleteParts(segmentId)
+
+			expect(await db.collection(COLLECTION_NAME).countDocuments()).toBe(0)
+		})
+
+		it('deletes multiple parts successfully', async () => {
+			const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+			const segmentId: string = 'someSegmentId'
+			const parts: Part[] = [createPart({ segmentId: segmentId }), createPart({ segmentId: segmentId })]
+			await testDatabase.populateDatabaseWithParts(parts)
+			const db: Db = testDatabase.getDatabase()
+
+			when(mongoConverter.convertParts(anything())).thenReturn(parts)
+			const testee: PartRepository = await createTestee(db, {
+				mongoConverter: mongoConverter,
+			})
+
+			await testee.deleteParts(segmentId)
+
+			expect(await db.collection(COLLECTION_NAME).countDocuments()).toBe(0)
+		})
+
+		// eslint-disable-next-line jest/expect-expect
+		it('calls deletion of pieces, matching amount of parts', async () => {
+			const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+			const pieceRepository: PieceRepository = mock<PieceRepository>()
+			const segmentId: string = 'someSegmentId'
+			const parts: Part[] = [createPart({ segmentId: segmentId }), createPart({ segmentId: segmentId })]
+			const pieces: Piece[] = [createPiece({}), createPiece({}), createPiece({})]
+			await testDatabase.populateDatabaseWithParts(parts)
+			const db: Db = testDatabase.getDatabase()
+
+			when(mongoConverter.convertParts(anything())).thenReturn(parts)
+			when(pieceRepository.getPieces(anything())).thenResolve(pieces)
+			const testee: PartRepository = await createTestee(db, {
+				mongoConverter: mongoConverter,
+				pieceRepository: pieceRepository,
+			})
+
+			await testee.deleteParts(segmentId)
+
+			verify(pieceRepository.deletePieces(anyString())).times(parts.length)
+		})
+
+		it('throws exception, when nonexistent segmentId is given', async () => {
+			const expectedErrorMessageFragment: string = 'Expected to delete one or more parts'
+			const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+			const nonExistingId: string = 'nonExistingId'
+			const part: Part = createPart({})
+			await testDatabase.populateDatabaseWithParts([part])
+			const db = testDatabase.getDatabase()
+
+			when(mongoConverter.convertParts(anything())).thenReturn([])
+			const testee: PartRepository = await createTestee(db, {
+				mongoConverter: mongoConverter,
+			})
+
+			expect.assertions(2)
+			try {
+				await testee.deleteParts(nonExistingId)
+			} catch (error) {
+				// It isn't conditional, as the test will fail, if not hit, due to the 'expect.assertions(2)'
```

Comment:
You don't need to write this sentence over and over again. You don't even need to write it once :)

Reply:
Have removed all but one of those comments.